src/water_cmd.cpp

```diff
@@ -1277,7 +1280,7 @@ void ConvertGroundTilesIntoWaterTiles()
 					FOR_EACH_SET_BIT(dir, _flood_from_dirs[slope & ~SLOPE_STEEP]) {
 						TileIndex dest = TileAddByDir(tile, (Direction)dir);
 						Slope slope_dest = GetTileSlope(dest) & ~SLOPE_STEEP;
-						if (slope_dest == SLOPE_FLAT || IsSlopeWithOneCornerRaised(slope_dest)) {
+						if (slope_dest == SLOPE_FLAT || IsSlopeWithOneCornerRaised(slope_dest) || IsTileType(dest, MP_VOID)) {
```

Comment:
Void tiles also have slopes, and `dest` could be a void tile. But it may not always have the desired slope. To comply with the new flood behaviour of MP_VOID tiles, it does not matter which slope `dest` have. It floods from the void.

Reply:
Just noticed something interesting. This is what `ConvertGroundTilesIntoWaterTiles`do if `RunTileLoop`never happens, in master: 
![screenshot3](https://user-images.githubusercontent.com/43006711/104031714-0210e180-51c5-11eb-88b4-493755553e83.png)
It already makes shores of those tiles adjacent to void tiles (except a few corner cases). It means it is `RunTileLoop` that dry them, that's why we never get to see them in this state in a real game. This also means, the original `ConvertGroundTilesIntoWaterTiles` isn't consistent, it's bugged.

The few corner cases are these...
![screenshot5](https://user-images.githubusercontent.com/43006711/104036359-36879c00-51cb-11eb-91bb-7e42bc08e82f.png)
...which `|| IsTileType(dest, MP_VOID)` solve...
![screenshot4](https://user-images.githubusercontent.com/43006711/104036356-35ef0580-51cb-11eb-94eb-da012ed5c118.png)